src/Core/Form/IdentifiableObject/CommandBuilder/Product/Combination/DefaultCombinationCommandsBuilder.php

```diff
@@ -37,7 +37,7 @@ class DefaultCombinationCommandsBuilder implements CombinationCommandsBuilderInt
      */
     public function buildCommands(CombinationId $combinationId, array $formData): array
     {
-        if (empty($formData['header']['is_default'])) {
+        if (empty($formData['header']['is_default']) && empty($formData['is_default'])) {
```

Comment:
Maybe we can remove the first part of the condition. I'm not sure if it's still used. If someone have any information on that, let me know.

Reply:
So actually you don't need to modify this build (it's better to keep it as simple as possible). Instead, you should update this service https://github.com/PrestaShop/PrestaShop/blob/83b395cea0983de5e96e4372d6928370df8d9bf1/src/Core/Form/IdentifiableObject/DataFormatter/CombinationListFormDataFormatter.php#L43 and add the missing mapping for the default field (you will have to update the unit tests also I believe).

Reply:
Thanks @jolelievre. Done